modules/tinymce/src/themes/silver/main/ts/backstage/Backstage.ts

```diff
@@ -56,9 +57,23 @@ const init = (lazySinks: { popup: () => Result<AlloyComponent, string>; dialog:
     icons: () => editor.ui.registry.getAll().icons,
     menuItems: () => editor.ui.registry.getAll().menuItems,
     translate: I18n.translate,
-    isDisabled: () => editor.mode.isReadOnly() || !editor.ui.isEnabled(),
+    isDisabled: () => !editor.ui.isEnabled(),
     getOption: editor.options.get,
-    tooltips: TooltipsBackstage(lazySinks.dialog)
+    tooltips: TooltipsBackstage(lazySinks.dialog),
+    checkUiComponentContext: (specContext: string) => {
+      const [ key, value = '' ] = specContext.split(':');
+      const contexts = editor.ui.registry.getAll().contexts;
+      const enabledInContext = Obj.get(contexts, key)
+        .fold(
+          // Fallback to 'mode:design' if key is not found
+          () => Obj.get(contexts, 'mode').map((pred) => pred('design')).getOr(false),
+          (pred) => Strings.startsWith(value, '!') ? !pred(value.slice(1)) : pred(value)
```

Comment:
I didn't follow through on the PR where this was changed, sorry, so I don't remember exactly whether I suggested this and if there was a reason why it wasn't used.

```suggestion
          (pred) => value.charAt(0) === '!' ? !pred(value.slice(1)) : pred(value)
```
Since we are in hot code this will be much faster; we know we're only checking one character. `startsWith` is a custom implementation, it doesn't currently use the native `startsWith`, so it's doing quite a bit of unnecessary work for this simple check.

Reply:
It makes sense, I remember that was also suggested, but I found something related to Katamari `Strings`. And yeah, since this is used everywhere, this must have less overhead 👍